pkg/apis/core/taint.go

```diff
@@ -27,10 +27,8 @@ func (t *Taint) MatchTaint(taintToMatch Taint) bool {
 	return t.Key == taintToMatch.Key && t.Effect == taintToMatch.Effect
 }
 
-// taint.ToString() converts taint struct to string in format key=value:effect or key:effect.
+// taint.ToString() converts taint struct to string in format key=value:effect.
+// The value is optional and may be an empty string.
 func (t *Taint) ToString() string {
-	if len(t.Value) == 0 {
-		return fmt.Sprintf("%v:%v", t.Key, t.Effect)
```

Comment:
I'd actually have gone the other way and let parseTaint parse this format

Reply:
Thanks for reviewing, @liggitt. Can you please share why you'd let `parseTaint` parse this format? 

Are you concerned the change breaks compatibility for external consumers of `ToString()` that it to generate both formats? I agree this is a problem, but I'd like to understand it better.

Changing `parseTaint` can also break compatibility for external consumers, since `parseTaint` is used in `kubectl taint` and `kubelet --register-with-taints`, and external tools may rely on the current (more conservative) parsing to reject the format `key:effect`.

(As an aside, modifying `parseTaint` is a larger change. There are two copies of `parseTaint` that need to be updated, in `pkg/util/taints` and `cmd/kubectl/cmd/taint`. We'd also need to update examples documentation for `kubectl taint` and `kubelet --register-with-taints`.)

Reply:
> Can you please share why you'd let `parseTaint` parse this format?

Parsing the unambiguous stringification of the Taint object seems worthwhile. `<key>:<value>` also reads nicer than `<key>=:<value>`

> Changing `parseTaint` can also break compatibility for external consumers, since `parseTaint` is used in `kubectl taint` and `kubelet --register-with-taints`, and external tools may rely on the current (more conservative) parsing to reject the format `key:effect`.

If it can be parsed unambiguously to a well-formed v1.Taint object, then I don't think there is an issue. All existing inputs would still work, and additional inputs would now work and also produce well-formed API structs.

Reply:
Agreed. I'll update `parseTaint` implementation and amend this PR.

Reply:
I filed https://github.com/kubernetes/kubernetes/pull/74159 for the `parserTaint` changes. In order to preserve the commit that we discussed above, I am going to close this PR.